### DISCUSSION TITLE
Export React Stately Selection type

### DISCUSSION THREAD

**Discussion by NguyenDa18** [adobe/react-spectrum#3995] [General]
React Stately Selection datatype is the only thing not exported in packages/@react-stately/selection/src/index.ts and provides value for typing selections in React Spectrum. React Spectrum TableViews use the React Stately Selection datatype: checking the selection parameter of onSelectionChange, it can be observed that the selection is an extended Set object with additional keys currentKey and anchorKey. Because the React Stately Selection is not a currently exported type, the additional keys information is lost when a selection is typed as just the standard `@react-types/shared` Selection type.

**Comment by LFDanLu** [comment]
For posterity: this turned out to be something in an intermediary project, the type for the `onSelectionChange` should match the `Selection` type defined [here](https://github.com/adobe/react-spectrum/blob/97ff9f95d91befaf87251e52ea484f81daae8f3a/packages/%40react-types/shared/src/selection.d.ts#L28)
